packages/skills/carpark_client/dialogues.py

```diff
@@ -21,135 +21,35 @@
-        return result
-
-    def is_expecting_matching_accept(self) -> bool:
-        """
-        Check whether the dialogue is expecting a matching accept.
-
-        :return: True if yes, False otherwise.
-        """
-        last_sent_message = self._outgoing_messages[-1] if len(self._outgoing_messages) > 0 else None  # type: Optional[Message]
-        result = (last_sent_message is not None) and last_sent_message.get("performative") == FIPAMessage.Performative.ACCEPT
-        return result
-
-    def is_expecting_cfp_decline(self) -> bool:
-        """
-        Check whether the dialogue is expecting an decline following a cfp.
-
-        :return: True if yes, False otherwise.
-        """
-        last_sent_message = self._outgoing_messages[-1] if len(self._outgoing_messages) > 0 else None  # type: Optional[Message]
-        result = (last_sent_message is not None) and last_sent_message.get("performative") == FIPAMessage.Performative.CFP
-        return result
-
-    def is_expecting_accept_decline(self) -> bool:
-        """
-        Check whether the dialogue is expecting an decline following an accept.
-
-        :return: True if yes, False otherwise.
-        """
-        last_sent_message = self._outgoing_messages[-1] if len(self._outgoing_messages) > 0 else None  # type: Optional[Message]
-        result = (last_sent_message is not None) and last_sent_message.get("performative") == FIPAMessage.Performative.ACCEPT
-        return result
-
-    def is_expecting_inform(self) -> bool:
-        """
-        Check whether the dialogue is expecting an inform.
-
-        :return: True if yes, False otherwise.
-        """
-        last_sent_message = self._outgoing_messages[-1] if len(self._outgoing_messages) > 0 else None  # type: Optional[Message]
-        result = (last_sent_message is not None) and last_sent_message.get("performative") == FIPAMessage.Performative.INFORM
-        return result
-
-
-class DialogueStats(object):
-    """Class to handle statistics on the negotiation."""
-
-    def __init__(self) -> None:
-        """Initialize a StatsManager."""
-        self._self_initiated = {Dialogue.EndState.SUCCESSFUL: 0,
-                                Dialogue.EndState.DECLINED_CFP: 0,
-                                Dialogue.EndState.DECLINED_ACCEPT: 0}  # type: Dict[Dialogue.EndState, int]
-
-    @property
-    def self_initiated(self) -> Dict[Dialogue.EndState, int]:
-        """Get the stats dictionary on self initiated dialogues."""
-        return self._self_initiated
-
-    def add_dialogue_endstate(self, end_state: Dialogue.EndState) -> None:
-        """
-        Add dialogue endstate stats.
-
-        :param end_state: the end state of the dialogue
-        :return: None
-        """
-        self._self_initiated[end_state] += 1
-
 
-class Dialogues(SharedClass):
+class Dialogues(SharedClass, FIPADialogues):
```

Comment:
For this to be a shared class it needs to be instantiated here. Alternatively, it could be made an attribute of a handler for instance.